tooling/noir_js_backend_barretenberg/test/public_input_deflattening.test.ts

```diff
@@ -0,0 +1,93 @@
+import { Abi } from '@noir-lang/types';
+import { expect } from 'chai';
+import { flattenPublicInputsAsArray, deflattenPublicInputs, flattenPublicInputs } from '../src/public_inputs.js';
+
+const abi: Abi = {
+  parameters: [
+    {
```

Comment:
This is okay for now (not blocking) -- if we change the ABI format it could be quite problematic

Reply:
Yeah, I agree. I want to get to a point where we're automatically rebuilding these artifacts in CI before we run the tests / checking consistency at some point soon as it's too easy for things to become stale.